spec/std/http/web_socket_spec.cr

```diff
@@ -293,6 +293,84 @@ describe HTTP::WebSocket do
     end
   end
 
+  it "negotiates over HTTP correctly" do
+    port_chan = Channel(Int32).new
+
+    spawn do
+      http_ref = nil
+      ws_handler = HTTP::WebSocketHandler.new do |ws, ctx|
+        ctx.request.path.should eq("/")
+        ws.on_message do |str|
+          ws.send("pong #{str}")
+        end
+
+        ws.on_close do
+          http_ref.not_nil!.close
+        end
+        ws.run
+      end
+
+      http_server = http_ref = HTTP::Server.new(0, [ws_handler])
+      http_server.bind
+      port_chan.send(http_server.port)
+      http_server.listen
+    end
+
+    listen_port = port_chan.receive
+
+    ws2 = HTTP::WebSocket.new("ws://127.0.0.1:#{listen_port}")
```

Comment:
I consider it a bug that I have to use `ws2` to differentiate this variable from `ws` above, as *that* ws is introduced as a block parameter, which should introduce an entirely new variable.